front/pages/api/poke/workspaces/[wId]/data_sources/[name]/index.ts

```diff
@@ -96,13 +96,17 @@ async function handler(
           true
         );
         if (connDeleteRes.isErr()) {
-          return apiError(req, res, {
-            status_code: 500,
-            api_error: {
-              type: "internal_server_error",
-              message: `Error deleting connector: ${connDeleteRes.error.error.message}`,
-            },
-          });
+          // If we get a not found we proceed with the deletion of the data source. This will enable
+          // us to retry deletion of the data source if it fails at the Core level.
```

Comment:
Are you sure we will be able to retry? Cause then we delete the datasource and I'm expecting that we won't have the button to delete anymore if we deleted the datasource?

Reply:
We will, if the core data source deletion fails, we delete the connector but not the core data source nor the front one.